io/plugins/eu.esdihumboldt.hale.io.xls/src/eu/esdihumboldt/hale/io/xls/writer/XLSInstanceWriter.java

```diff
@@ -99,55 +99,60 @@ else if (getContentType().getId().equals("eu.esdihumboldt.hale.io.xls.xlsx")) {
-			return reporter;
-		}
+		// use ResourceIterator<Instance> in a try block because is closable -
+		// avoid infinite
+		// cleaning project after exporting data
+		try (ResourceIterator<Instance> instanceIterator = instances.iterator();) {
+			Instance instance = null;
+			try {
+				instance = instanceIterator.next();
+			} catch (NoSuchElementException e) {
+				reporter.error(
+						new IOMessageImpl("There are no instances for the selected type.", e));
+				return reporter;
+			}
 
-		List<Instance> remainingInstances = new ArrayList<Instance>();
+			List<Instance> remainingInstances = new ArrayList<Instance>();
 
-		headerRowStrings = new ArrayList<String>();
+			headerRowStrings = new ArrayList<String>();
 
-		boolean useSchema = getParameter(InstanceTableIOConstants.USE_SCHEMA).as(Boolean.class,
-				true);
+			boolean useSchema = getParameter(InstanceTableIOConstants.USE_SCHEMA).as(Boolean.class,
+					true);
 
-		// all instances with equal type definitions are stored in an extra
-		// sheet
-		TypeDefinition definition = instance.getDefinition();
+			// all instances with equal type definitions are stored in an extra
+			// sheet
+			TypeDefinition definition = instance.getDefinition();
 
-		Sheet sheet = wb.createSheet(definition.getDisplayName());
-		Row headerRow = sheet.createRow(0);
-		int rowNum = 1;
-		Row row = sheet.createRow(rowNum++);
-		writeRow(row,
-				super.getPropertyMap(instance, headerRowStrings, useSchema, solveNestedProperties));
+			Sheet sheet = wb.createSheet(definition.getDisplayName());
+			Row headerRow = sheet.createRow(0);
+			int rowNum = 1;
+			Row row = sheet.createRow(rowNum++);
+			writeRow(row, super.getPropertyMap(instance, headerRowStrings, useSchema,
+					solveNestedProperties));
 
-		while (instanceIterator.hasNext()) {
-			Instance nextInst = instanceIterator.next();
-			if (nextInst.getDefinition().equals(definition)) {
-				row = sheet.createRow(rowNum++);
-				writeRow(row, super.getPropertyMap(nextInst, headerRowStrings, useSchema,
-						solveNestedProperties));
+			while (instanceIterator.hasNext()) {
+				Instance nextInst = instanceIterator.next();
+				if (nextInst.getDefinition().equals(definition)) {
+					row = sheet.createRow(rowNum++);
+					writeRow(row, super.getPropertyMap(nextInst, headerRowStrings, useSchema,
+							solveNestedProperties));
+				}
+				else
+					remainingInstances.add(nextInst);
 			}
-			else
-				remainingInstances.add(nextInst);
-		}
-		writeHeaderRow(headerRow, headerRowStrings);
-		setCellStyle(sheet, headerRowStrings.size());
-		resizeSheet(sheet);
+			writeHeaderRow(headerRow, headerRowStrings);
+			setCellStyle(sheet, headerRowStrings.size());
+			resizeSheet(sheet);
 
-		// write file
-		FileOutputStream out = new FileOutputStream(getTarget().getLocation().getPath());
-		wb.write(out);
-		out.close();
+			// write file
+			FileOutputStream out = new FileOutputStream(getTarget().getLocation().getPath());
```

Comment:
I think you can put this `FileOutputStream out = new FileOutputStream(getTarget().getLocation().getPath());` also in a `try(...)` block. As it also closes the file stream resource.

Reply:
Right!